modules/speech-to-text-impl/src/test/java/org/opencastproject/speechtotext/impl/SpeechToTextServiceImplTest.java

```diff
@@ -0,0 +1,163 @@
+
+    Workspace workspace = EasyMock.createMock(Workspace.class);
+    final String directory = testFolder.newFolder().getAbsolutePath();
+    EasyMock.expect(workspace.rootDirectory()).andReturn(directory).anyTimes();
+
+    EasyMock.expect(workspace.get(EasyMock.anyObject()))
+            .andReturn(new File(getClass().getResource("/speech_to_text_test.mp4").toURI())).anyTimes();
+
+    Field field = speechToTextService.getClass().getDeclaredField("selectedEngine");
+    field.setAccessible(true);
+    field.set(speechToTextService, voskEngine);
+
+    final Capture<String> collection = EasyMock.newCapture();
+    final Capture<String> name = EasyMock.newCapture();
+    final Capture<InputStream> in = EasyMock.newCapture();
+    EasyMock.expect(workspace.putInCollection(capture(collection), capture(name), capture(in))).andAnswer(() -> {
+      File output = new File(directory, "speech_to_text_test.vtt");
+      FileUtils.copyInputStreamToFile(in.getValue(), output);
+      return output.toURI();
+    }).once();
+
+    final Capture<String> collectionID = EasyMock.newCapture();
+    final Capture<String> fileName = EasyMock.newCapture();
+    EasyMock.expect(workspace.getCollectionURI(capture(collectionID), capture(fileName))).andAnswer(() -> {
+      File tmpVttFile = new File(directory + "/collection/" + collectionID.getValue() + "/" + fileName.getValue());
+      return tmpVttFile.toURI();
+    }).anyTimes();
+
+    workspace.delete(EasyMock.anyObject());
+
+    // Finish setting up the mocks
+    EasyMock.replay(bc, cc, workspace);
+
+    ServiceRegistry serviceRegistry = EasyMock.createMock(ServiceRegistry.class);
+    final Capture<String> type = EasyMock.newCapture();
+    final Capture<String> operation = EasyMock.newCapture();
+    final Capture<List<String>> args = EasyMock.newCapture();
+    EasyMock.expect(serviceRegistry.createJob(capture(type), capture(operation), capture(args), EasyMock.anyFloat()))
+            .andAnswer(() -> {
+              Job job = new JobImpl(0);
+              job.setJobType(type.getValue());
+              job.setOperation(operation.getValue());
+              job.setArguments(args.getValue());
+              job.setPayload(speechToTextService.process(job));
+              return job;
+            }).anyTimes();
+    EasyMock.replay(serviceRegistry);
+
+    speechToTextService.setServiceRegistry(serviceRegistry);
+    speechToTextService.setWorkspace(workspace);
+  }
+
+  @Test
+  public void testVoskSpeechToText() throws Exception {
+    URI videoUri = getClass().getResource("/speech_to_text_test.mp4").toURI();
+    Job job = speechToTextService.transcribe(videoUri, "eng");
+    File output = new File(new URI(job.getPayload()));
+    StringBuilder textFromSpeech = new StringBuilder();
+    try (Stream<String> stream = Files.lines(Paths.get(output.getPath()), StandardCharsets.UTF_8)) {
+      stream.forEach(s -> textFromSpeech.append(s).append("\n"));
```

Comment:
Isn't this just a more complex way of
```java
Files.lines(Paths.get(output.getPath()), StandardCharsets.UTF_8).collect(Collectors.joining("\n"));
```

I'm not actually sure what this is supposed to test except for the possible `IOException` which, when thrown, would already let the test fail by itself, wouldn't it?